crates/matrix-sdk/tests/integration/client.rs

```diff
@@ -412,21 +426,35 @@ async fn join_room_by_id_or_alias() {
     Mock::given(method("POST"))
         .and(path_regex(r"^/_matrix/client/r0/join/"))
         .and(header("authorization", "Bearer 1234"))
-        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::ROOM_ID))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_json(json!({"room_id":  *test_json::DEFAULT_SYNC_ROOM_ID})),
+        )
         .mount(&server)
         .await;
 
-    let room_id = room_id!("!testroom:example.org").into();
+    mock_sync(&server, &*test_json::SYNC, None).await;
+
+    let room_id_or_alias: &RoomOrAliasId =
+        <&RoomOrAliasId>::try_from(*test_json::DEFAULT_SYNC_ROOM_ID).unwrap();
```

Comment:
Type annotation is redundant with turbofish before `try_from`.

```suggestion
    let room_id_or_alias =
        <&RoomOrAliasId>::try_from(*test_json::DEFAULT_SYNC_ROOM_ID).unwrap();
```